main.go

```diff
@@ -33,19 +48,56 @@ func main() {
 		_ = resp.Body.Close()
 	}()
 
+	// Check that the domain specified by the user exists in GoDaddy
+	domainExists := 0
 	for _, zone := range zones {
-		if zone.Status == "ACTIVE" {
-			log.Println(zone.Domain)
+		if strings.EqualFold(zone.Status, "active") && strings.EqualFold(zone.Domain, *userDomainName) {
+			domainExists = 1
+			log.Println("Domain", zone.Domain, "matches user domain", *userDomainName)
 		}
 	}
 
-	records, _, err := apiClient.V1Api.RecordGet(ctx, "lwts.org", "", "", nil)
+	if domainExists != 1 {
+		log.Println("Domain", *userDomainName, "does not match any domains returned from the API")
+		os.Exit(1)
+	}
+
+	var recordData map[string]string
+	recordData = make(map[string]string)
```

Comment:
if you change this to `recordData := make(map[string]string)` you can remove the `var` line above it.